artifactory/utils/pip/setuppy.go

```diff
@@ -43,57 +45,52 @@ func getProjectNameFromFileContent(content []byte) (string, error) {
 // Run egg-info command on setup.py, the command generates metadata files.
 // Return the content of the 'PKG-INFO' file.
 func getEgginfoPkginfoContent(setuppyFilePath, pythonExecutablePath string) ([]byte, error) {
-	tempDirPath, err := fileutils.CreateTempDir()
+	eggBase, err := fileutils.CreateTempDir()
 	if err != nil {
 		return nil, err
 	}
-	defer fileutils.RemoveTempDir(tempDirPath)
+	defer func() {
+		e := fileutils.RemoveTempDir(eggBase)
+		if err == nil {
+			err = e
+		}
+	}()
 
-	// Run python egg_info command.
-	egginfoOutput, err := executeEgginfoCommandWithOutput(pythonExecutablePath, setuppyFilePath, tempDirPath)
-	if err != nil {
+	// Run python 'egg_info --egg-base <eggBase>' command.
+	if err = executeEgginfo(pythonExecutablePath, setuppyFilePath, eggBase); err != nil {
```

Comment:
could be replaced by 
`if err = exec.Command(pythonExecutablePath, setuppyFilePath, "egg_info", "--egg-base", tempDirPath).Run() ....`